DB/src/main/java/io/deephaven/db/v2/sources/chunkcolumnsource/CharChunkColumnSource.java

```diff
@@ -0,0 +1,133 @@
+                final int offset = (int)(firstKey - offsets.get(firstChunk));
+                final int length = orderedKeys.intSize();
+                final CharChunk<? extends Attributes.Values> charChunk = data.get(firstChunk);
+                if (offset == 0 && length == charChunk.size()) {
+                    return charChunk;
+                }
+                return ((ChunkGetContext)context).resettableCharChunk.resetFromChunk(charChunk, offset, length);
+            }
+        }
+        return getChunkByFilling(context, orderedKeys);
+    }
+
+    @Override
+    public void fillChunk(@NotNull FillContext context, @NotNull WritableChunk<? super Attributes.Values> destination, @NotNull OrderedKeys orderedKeys) {
+        final MutableInt startChunkIndex = new MutableInt(0);
+        final MutableInt destinationOffset = new MutableInt(0);
+        orderedKeys.forAllLongRanges((s, e) -> {
+            while (s <= e) {
+                final int chunkIndex = getChunkIndex(s, startChunkIndex.intValue());
+                final int offsetWithinChunk = (int)(s - offsets.get(chunkIndex));
+                Assert.geqZero(offsetWithinChunk, "offsetWithinChunk");
+                final CharChunk<? extends Attributes.Values> charChunk = data.get(chunkIndex);
+                final int chunkSize = charChunk.size();
+                final long rangeLength = e - s + 1;
+                final int chunkRemaining = chunkSize - offsetWithinChunk;
+                final int length = rangeLength > chunkRemaining ? chunkRemaining : (int) rangeLength;
+                Assert.gtZero(length, "length");
+                destination.copyFromChunk(charChunk, offsetWithinChunk, destinationOffset.intValue(), length);
+                destinationOffset.add(length);
+                startChunkIndex.setValue(chunkIndex + 1);
+                s += length;
+            }
+        });
+    }
+
+    @Override
+    public void fillPrevChunk(@NotNull FillContext context, @NotNull WritableChunk<? super Attributes.Values> destination, @NotNull OrderedKeys orderedKeys) {
+        // immutable, so we can delegate to fill
+        fillChunk(context, destination, orderedKeys);
+    }
+
+    private int getChunkIndex(final long start) {
+        return getChunkIndex(start, 0);
+    }
+
+    private int getChunkIndex(final long start, final int startIndex) {
+        int index = offsets.binarySearch(start, startIndex, offsets.size());
+        if (index < 0) {
+            index = -index - 2;
+        }
+        return index;
+    }
+
+    public void addChunk(final CharChunk<? extends Attributes.Values> chunk) {
+        data.add(chunk);
+        offsets.add(totalSize);
+        totalSize += chunk.size();
+    }
+
+    @Override
+    public void addChunk(final Chunk<? extends Attributes.Values> chunk) {
+        addChunk(chunk.asCharChunk());
+    }
+
+    @Override
+    public void clear() {
+        totalSize = 0;
+        data.clear();
+        offsets.clear();
```

Comment:
You want to use `offsets.reset()`, I think. `clear()` throws away the backing array.